src/core/initial_conditions.jl

```diff
@@ -99,6 +99,21 @@ function calculate_ic_quantity(
     return var_value
 end
 
+function calculate_ic_quantity(
+    initial_condition_key::ICKey{DeviceEnergy, T},
```

Comment:
EnergyLevel  or energy quantity makes more sense

Reply:
you want me to change DeviceEnergy to EnergyLevel for  InitialConditionType ?

Reply:
done